src/OrchardCore.Modules/OrchardCore.AuditTrail/Controllers/AdminController.cs

```diff
@@ -0,0 +1,111 @@
+
+
+        public AdminController(
+            ISiteService siteService,
+            IShapeFactory shapeFactory,
+            IAuditTrailManager auditTrailManager,
+            IUpdateModelAccessor updateModelAccessor,
+            IAuthorizationService authorizationService,
+            IAuditTrailEventDisplayManager auditTrailEventDisplayManager)
+        {
+            _siteService = siteService;
+            _shapeFactory = shapeFactory;
+            _auditTrailManager = auditTrailManager;
+            _updateModelAccessor = updateModelAccessor;
+            _authorizationService = authorizationService;
+            _auditTrailEventDisplayManager = auditTrailEventDisplayManager;
+        }
+
+
+        public async Task<ActionResult> Index(PagerParameters pagerParameters, AuditTrailOrderBy? orderBy = null)
+        {
+            if (!await _authorizationService.AuthorizeAsync(User, AuditTrailPermissions.ViewAuditTrail))
+            {
+                return Forbid();
+            }
+
+            var siteSettings = await _siteService.GetSiteSettingsAsync();
+            var pager = new Pager(pagerParameters, siteSettings.PageSize);
+            var filters = Filters.From(QueryHelpers.ParseQuery(Request.QueryString.Value), _updateModelAccessor.ModelUpdater);
+            var searchResult = 
+                await _auditTrailManager.GetAuditTrailEventsAsync(pager.Page, pager.PageSize, filters, orderBy ?? AuditTrailOrderBy.DateDescending);
+
+            if (!_updateModelAccessor.ModelUpdater.ModelState.IsValid)
+            {
+                searchResult.AuditTrailEvents = Enumerable.Empty<AuditTrailEvent>();
+            }
+
+            var pagerShape = (await _shapeFactory.New.Pager(pager)).TotalItemCount(searchResult.TotalCount);
+
+            var eventDescriptors = _auditTrailManager.DescribeCategories()
+                .SelectMany(categoryDescriptor => categoryDescriptor.Events)
+                .ToDictionary(eventDescriptor => eventDescriptor.FullEventName);
+
+            var auditTrailEventsSummaryViewModel = searchResult.AuditTrailEvents.Select(auditTrailEvent =>
+            {
+                var eventDescriptor = eventDescriptors.ContainsKey(auditTrailEvent.FullEventName) ? 
+                    eventDescriptors[auditTrailEvent.FullEventName] : AuditTrailEventDescriptor.Basic(auditTrailEvent);
+                return new AuditTrailEventSummaryViewModel
+                {
+                    AuditTrailEvent = auditTrailEvent,
+                    EventDescriptor = eventDescriptor,
+                    CategoryDescriptor = eventDescriptor.CategoryDescriptor,
+                    AdditionalColumnsShapes = _auditTrailEventDisplayManager.BuildAdditionalColumnsShapesAsync(auditTrailEvent),
+                    SummaryShape = _auditTrailEventDisplayManager.BuildDisplayAsync(auditTrailEvent, "SummaryAdmin"),
+                    ActionsShape = _auditTrailEventDisplayManager.BuildActionsAsync(auditTrailEvent, "SummaryAdmin")
```

Comment:
Could this `IAuditTrailEventDisplayManager` possibly be an `IDisplayManager<AuditTrailEvent>` with `IDisplayDrivers` and render to zones, rather than a whole new implementation with shape factories ?

After trying with shape providers and factories I did this for the content item list filters, and it seemed to work better, and be much simpler.

Reply:
To be able to roll this out so people can start using it, after which we can then refine it, Gábor opened an issue for it: https://github.com/OrchardCMS/OrchardCore/issues/8622